### PR TITLE
spine add stats and ES6 refactor

### PR DESCRIPTION
**Pull request by lodoyun**
**Description**
This PR implements:
- Spine is now updated to ES6.
- Spine is now a different object that is used by `runSpineClients`.
- Stats and Stream status can now be gathered when running a Spine Session.

You can specify an array of stat keys, a stat gathering interval and total number of intervals in `spineClientsConfig.json` The result will be written to a file that can be specified as a command line parameter for `runSpineClients`. The result will always be the output from the last stats gathering.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: spine/simpleNativeConnection.js [hunk 1]
```diff
@@ -1,157 +1,174 @@
-'use strict';
 process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';  // We need this for testing with self signed certs
 
-var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest,
-    nodeUrl = require ('url'),
-    Erizo = require('./erizofc'),
-    NativeStream = require ('./NativeStream.js'),
-    NativeStack = require ('./NativeStack.js');
+const XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest; // eslint-disable-line import/no-extraneous-dependencies
```
Comment (jcague) [inline spine/simpleNativeConnection.js]: should we add a package.json in spine in the future?

File: spine/runSpineClients.js [hunk 2]
```diff
@@ -1,105 +1,127 @@
...
 
 console.log('Loading stream config file', streamConfig);
-streamConfig = require('./' + streamConfig);
-
-if (streamConfig.publishConfig){
-    var streamPublishConfig = {
-        publishConfig: streamConfig.publishConfig,
-        serverUrl: streamConfig.basicExampleUrl
-    };
+streamConfig = require(`./${streamConfig}`); // eslint-disable-line
```
Comment (jcague) [inline spine/runSpineClients.js]: there are other ways to read the config file better than require, we can fix it in the future.

File: spine/simpleNativeConnection.js [hunk 3]
```diff
@@ -1,157 +1,174 @@
...
+    room.disconnect();
+  };
+
+  that.getStats = () => {
+    const promises = [];
+    availableStreams.forEach((stream, streamId) => {
+      if (stream.pc && stream.pc.peerConnection && stream.pc.peerConnection.connected) {
+        promises.push(stream.pc.peerConnection.getStats());
+      } else {
+        log.info('No stream to ask for stats: ', streamId);
```
Comment (jcague) [inline spine/simpleNativeConnection.js]: log.warn?

File: spine/NativeStack.js [hunk 4]
```diff
@@ -1,85 +1,87 @@
...
+
+  that.pcConfig = {
+    iceServers: [],
+  };
+
+  if (config.iceServers !== undefined) {
+    that.pcConfig.iceServers = configuration.iceServers;
+  }
+
+  if (configuration.audio === undefined) {
```
Comment (jcague) [inline spine/NativeStack.js]: consider `configuration.audio = configuration.audio || false`

File: spine/NativeStack.js [hunk 5]
```diff
@@ -1,85 +1,87 @@
...
+    configuration.audio = false;
+  }
+
+  if (configuration.video === undefined) {
```
Comment (jcague) [inline spine/NativeStack.js]: consider `configuration.video = configuration.video || false`

File: .jshintignore [hunk 6]
```diff
@@ -1,7 +1,7 @@
 nuve/nuveClient/build
 nuve/nuveClient/dist
-erizo_controller/erizoClient
-spine/erizofc.js
+erizo_controller/erizoClient/**/*.js
+spine/**/*.js
```
Comment (lodoyun) [inline .jshintignore]: It helps when you have an editor configured with both eslint and jshint.
The idea is to create a gulp task or similar in spine to lint it, will do in a future PR.

File: spine/simpleNativeConnection.js (hunk 1 above)
Comment (lodoyun) [inline spine/simpleNativeConnection.js]: We should, i'd do it when we create a package.json for each of the parts of licode.

File: spine/runSpineClients.js (hunk 2 above)
Comment (lodoyun) [inline spine/runSpineClients.js]: 👍

**Pull request merged**